### PR TITLE
feat(source): add metrics for source

### LINKED ISSUE #3422: refine error message caused by internal kafka failures
Though this is indeed an internal error, the message is quite unfriendly to read. The error is user-facing. We should refine it and let users immediately know the cause and how to fix.

```
dev=> create materialized view ad_mv as select * from ad_source;
ERROR:  RPC error: GrpcStatus(Status { code: Internal, message: "internal error: Meta data fetch error: BrokerTransportFailure (Local: Broker transport failure)", metadata: MetadataMap { headers: {"content-type": "application/grpc", "risingwave-error-bin": "CAESX2ludGVybmFsIGVycm9yOiBNZXRhIGRhdGEgZmV0Y2ggZXJyb3I6IEJyb2tlclRyYW5zcG9ydEZhaWx1cmUgKExvY2FsOiBCcm9rZXIgdHJhbnNwb3J0IGZhaWx1cmUp", "date": "Thu, 23 Jun 2022 05:48:12 GMT", "content-length": "0"} }, source: None })
```

```
Internal Error: The connection to Kafka brokers ("127.0.0.1:29092") is broken, please check its availability.
```

### PR DESCRIPTION
**Pull request by tabVersion**
Signed-off-by: tabVersion <tabvision@bupt.icu>

I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

***PLEASE DO NOT LEAVE THIS EMPTY !!!***

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

* move `StreamingMetrics` to risingwave_common to solve cycle dependency
* refine parse error message and support skip wrong message
* add a counter metric for wrong message

- Summarize your change (**mandatory**)
- How does this PR work? Need a brief introduction for the changed logic (optional)
- Describe clearly one logical change and avoid lazy messages (optional)
- Describe any limitations of the current code (optional)
- Add the 'user-facing changes' label if your PR contains changes that are visible to users (optional)

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)

close #3422

### CODE REVIEW

File: src/common/src/monitor/streaming_stats.rs [hunk 1]
```diff
@@ -197,6 +198,14 @@ impl StreamingMetrics {
             register_histogram_vec_with_registry!(opts, &["actor_id", "wait_side"], registry)
                 .unwrap();
 
+        let source_parse_error_count = register_int_counter_vec_with_registry!(
```
Comment (skyzh) [inline src/common/src/monitor/streaming_stats.rs:201]: I have no idea why developers are addicted to move everything to common crate. Do you want to wait for 5 minutes for a release build when you just add a simple counter in prometheus?

Comment (skyzh) [reply src/common/src/monitor/streaming_stats.rs:201]: StreamMetrics should only be used in streaming module. You can always create a SourceMetrics for connectors.

**Review by skyzh — COMMENTED** [review]
It seems that this metrics is not added to dashboard? If you want it to be displayed on Grafana dashboard, please update `risingwave-dashboard.py` and generate the new dashboard.

Rest LGTM, and please don't make common larger and larger. It's on the critical path of compiling everything, and should be as little as possible.

File: src/common/src/monitor/streaming_stats.rs (hunk 1 above)
Comment (tabVersion) [reply src/common/src/monitor/streaming_stats.rs:201]: Sure. I will implement a new Metric type. will fix.

**Comment by neverchanje** [comment]
Any progress in this PR?

**Comment by tabVersion** [comment]
> Any progress in this PR?

I have forgotten the pr, the feature has been done in #3505.
close for duplicate

**Pull request closed without merging**